cmake/toolchain/xcc/Kconfig

```diff
@@ -0,0 +1,7 @@
+# xcc Toolchain
+
+# Copyright (c) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config TOOLCHAIN_XCC
```

Comment:
an FYI there's a PR opened which changes the name of `xcc` and `xcc-clang` to more proper `xt-xcc` and `xt-clang`.
https://github.com/zephyrproject-rtos/zephyr/pull/54836

maybe wait for that change before introducing new Kconfig setting names which are based on toolchain names.

Reply:
I'm aware of that PR and we will fix up either this PR or that one depending on which goes in first.

Reply:
xcc didn't change, just xcc-clang -> xt-xcc.